conda/models/version.py

```diff
@@ -637,6 +645,10 @@ def merge(self, other):
                              % (self.raw_value, other.raw_value))
         return self.raw_value
 
+    def union(self, other):
```

Comment:
Looks like this union function is defined a couple times. Can if be extracted into a utils module or something and shared?

Reply:
yep.  I wouldn't bother reading too much into this code right now, just wanted to share the ideas, since they overlap with what you're doing.